Feature/queuelimits

bldrvnlw opened the pull request:
This contains the permissions implementation based on laravel-permission from Spatie. As well as permissions (users, roles and permissions functionality) it contains som overlap with the recently added Lowercase rule as follows:

1. The base permissions have been centralized in the app\User class and are retrieved with getValidationRules. This is in order to provide a common starting point for validation in Register and Modify of website user accounts.
2. Rules are applied using UserModifyRequest and UserRegistrationRequest classes (this is AFAIK Laravel best practice)
3. The app\User class has aan added mutator "SetPasswordAttribute" used to ensure that the password is always bcrypted - this removed the need to do this in the individual controllers either Modify or register.

Apologies for the long list of commits - if needs be I can do a squash to simplify the history. Essentially only the stuff from the last couple of weeks is relevant.

Add fix for issue with modify user - this replaces PR#14

Commit:
Running independent FUMA (with added debugbar)

Commit:
up to step in 53 version switch

Commit:
Working 5.3 except logout

Commit:
Fix login using POST and redirect correctly

Commit:
After jobs testing in Laravel 5.4

Commit:
Fix error in $this->$jobID

Commit:
Initial user roles and permissions commit

Commit:
Role Permission and User editing first steps

Commit:
Users, Roles, Permissions now functioning

Commit:
Test check working

Commit:
Job timeouts functioning

Commit:
Updated to laravel 5.5 PHP 7.3

Commit:
Upgrade to Laravel 5.6

Commit:
Fix the routing cache 5.7, run the update.

Commit:
Updated to 5.8.38. Bugs: count in jobs, missing jQuery

Commit:
The env is called QUEUE_CONNECTION now

Commit:
Refactor snp2geneProcess

Commit:
Fixed the error handling, timeout in cell, genemap

Commit:
Add a user example to the roles seeder

bldrvnlw committed:
Merge branch 'master' into feature/queuelimits

bldrvnlw committed:
Modify controllers and blade templates

Correct error handling
Centeralized validation rules
Using FormRequest
Also make the migration tables match current master

bldrvnlw committed:
Make seeds directory persistent

bldrvnlw committed:
Fix Permissions blade header

and correct type for analytics script

bldrvnlw committed:
Always preserve composer.lock on merge

composer.lock should be regenerated after a merge using composer update

bldrvnlw committed:
manual sync with latest master

bldrvnlw committed:
Missing route for FAQ

bldrvnlw committed:
Fix typo in blade layout

bldrvnlw committed:
Fix exception handler (function failed) signature

bldrvnlw committed:
Adding updated composer.lock

bldrvnlw committed:
Unique email ignore  during user modification, fix

bldrvnlw committed:
Must use the Rule::unique syntax

bldrvnlw committed:
Merge branch 'master' into feature/queuelimits

tanyaphung reviewed (approved):
I tested it on tfumatest and it is working.

bldrvnlw committed:
Seed an Admin user using an email via an env. var.

Run the seeding as follows:

SEED_ADMIN_EMAIL=bob@goodmail.com php7.3 artisan db:seed --class=RolesAndPermissionsSeeder

Will create the Admin role (if it does not already exist)
and add it to the user with email bob@goodmail.com

bldrvnlw committed:
Merge branch 'feature/queuelimits' of https://github.com/vufuma/FUMA-webapp into feature/queuelimits
